### PR TITLE
Autoresume related error messages

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Adds errors so that users should not be able to hit edge case bugs. See linked issues.

@mvpatel2000 I think I can fix the symlink/formatting issue (CO-1272), but wanted to get these in now for your full pipeline testing.

# What issue(s) does this change relate to?
Closes [CO-1365](https://mosaicml.atlassian.net/browse/CO-1365?atlOrigin=eyJpIjoiMmRhNWViNjU4ZGMwNGM1MTgwMzUzZmMzMWZkOWRjZDMiLCJwIjoiaiJ9).
Relates to [CO-1270](https://mosaicml.atlassian.net/browse/CO-1270) and [CO-1272](https://mosaicml.atlassian.net/browse/CO-1272)

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -1014,6 +1014,15 @@ def __init__(
         # Logger
         self.logger = Logger(state=self.state, destinations=loggers)
 
+        if save_latest_filename is not None:
+            if any(
+                    isinstance(logger_destination, RemoteUploaderDownloader) and
+                    logger_destination.file_path_format_string != '{remote_file_name}'
+                    for logger_destination in self.logger.destinations):
```
Comment (mvpatel2000) [inline composer/trainer/trainer.py:1021]: nit: maybe move part of this into a separate variable? possibly the list comprehension part? it's a kinda nasty if statement thats a little hard to parse

File: composer/trainer/trainer.py [hunk 2]
```diff
@@ -1214,6 +1223,12 @@ def __init__(
                 raise ValueError(
                     'The `run_name` must be specified when using autoresume so Event.INIT is run with the correct run name.'
                 )
+            if any(
+                    isinstance(logger_destination, RemoteUploaderDownloader) and
+                    logger_destination._num_concurrent_uploads != 1 for logger_destination in self.logger.destinations):
```
Comment (mvpatel2000) [inline composer/trainer/trainer.py:1228]: same as above

File: composer/trainer/trainer.py [hunk 3]
```diff
@@ -1214,6 +1223,12 @@ def __init__(
...
+                raise ValueError(
+                    'Multiple concurrent uploads is not currently supported when using autoresume. Please set `num_concurrent_uploads` to 1 '
+                    'for all `RemoteUploaderDownloader` instances.')
```
Comment (mvpatel2000) [inline composer/trainer/trainer.py:1231]: should we just... remove functionality to set more than 1 worker for now? like keep class functionality, but delete argument? is there ever a case where >1 worker is preferred given risk of race conditions?

**Review by mvpatel2000 — APPROVED** [review]

File: composer/trainer/trainer.py (hunk 3 above)
Comment (dakinggg) [reply composer/trainer/trainer.py:1231]: It _only_ breaks the symlink. So I think we should leave it in and figure out the real fix.

File: composer/trainer/trainer.py (hunk 1 above)
Comment (dakinggg) [reply composer/trainer/trainer.py]: refactored, check and see if looks good (both cases)

**Review by eracah — APPROVED** [review]
LGTM
